quickwit/quickwit-search/src/root.rs

```diff
@@ -744,9 +781,21 @@ pub(crate) async fn search_partial_hits_phase(
         has_intermediate_aggregation_result = leaf_search_response.intermediate_aggregation_result.is_some(),
         "Merged leaf search response."
     );
+
+    if let Some(resource_stats) = &leaf_search_response.resource_stats {
+        if is_top_5pct_memory_intensive(
+            resource_stats.short_lived_cache_num_bytes,
+            resource_stats.split_num_docs,
+        ) {
+            // We log at most 5 times per minute.
+            quickwit_common::rate_limited_info!(limit_per_min=5, split_num_docs=resource_stats.split_num_docs, %search_request.query_ast, short_lived_cached_num_bytes=resource_stats.short_lived_cache_num_bytes, query=%search_request.query_ast, "memory intensive query");
+        }
```

Comment:
should we create a metric as well?

Reply:
I think an histogram with not too many bucket for general memory usage. might be nice. what do you think?

Reply:
sounds good! The stats struct also contains other metrics: warmup_microsecs, cpu_thread_pool_wait_microsecs, cpu_microsecs. Did you have any plans for those?

Reply:
As discussed, we will not add timing metrics for now and just add a histogram about the final size of individual short lived caches (note that `shortlived_cache` was already added as global storage metric)